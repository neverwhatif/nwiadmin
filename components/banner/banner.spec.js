import React from 'react';
import { shallow } from 'enzyme';

import { BannerComponent } from './index';

jest.mock('nwiadmin/services/me/meable');
jest.mock('nwiadmin/components/primarynav', () => 'PrimaryNav');
jest.mock('./bannerlogo', () => 'BannerLogo');
jest.mock('./mepanel', () => 'MePanel');

const me = {
    name: 'Lorem Ipsum',
};

describe('Banner', () => {
    const component = shallow(<BannerComponent />);
    const componentWithMe = shallow(<BannerComponent me={me} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentWithMe).toMatchSnapshot();
    });

    it('should render logo if no me', () => {
        expect(component.children().length).toBe(1);
        expect(component.find('BannerLogo').length).toBe(1);
    });

    it('should render nav and mepanel if me', () => {
        expect(componentWithMe.children().length).toBe(2); // 2 instead of 3 due to Fragment
        expect(componentWithMe.find('MePanel').length).toBe(1);
        expect(componentWithMe.find('MePanel').prop('data')).toEqual(me);
        expect(componentWithMe.find('PrimaryNav').length).toBe(1);
    });
});
