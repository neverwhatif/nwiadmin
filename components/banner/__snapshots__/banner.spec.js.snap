// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BannerComponent
    isError={false}
    me={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": null,
      "className": "root rootPink",
    },
    "ref": null,
    "rendered": null,
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "className": "root rootPink",
      },
      "ref": null,
      "rendered": null,
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Banner should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BannerComponent
    isError={false}
    me={
      Object {
        "name": "Lorem Ipsum",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <React.Fragment>
        <BannerLogo />
        <PrimaryNav />
        <MePanel
          data={
            Object {
              "name": "Lorem Ipsum",
            }
          }
        />
      </React.Fragment>,
      "className": "root rootPink",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <BannerLogo />,
          <PrimaryNav />,
          <MePanel
            data={
              Object {
                "name": "Lorem Ipsum",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "BannerLogo",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "PrimaryNav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "data": Object {
              "name": "Lorem Ipsum",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "MePanel",
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <React.Fragment>
          <BannerLogo />
          <PrimaryNav />
          <MePanel
            data={
              Object {
                "name": "Lorem Ipsum",
              }
            }
          />
        </React.Fragment>,
        "className": "root rootPink",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <BannerLogo />,
            <PrimaryNav />,
            <MePanel
              data={
                Object {
                  "name": "Lorem Ipsum",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "BannerLogo",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "PrimaryNav",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "data": Object {
                "name": "Lorem Ipsum",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "MePanel",
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
